lib/libshare/nfs.c

```diff
@@ -47,10 +54,41 @@ static sa_fstype_t *nfs_fstype;
 static int nfs_exportfs_temp_fd = -1;
 
 typedef int (*nfs_shareopt_callback_t)(const char *opt, const char *value,
-    void *cookie);
+	void *cookie);
 
 typedef int (*nfs_host_callback_t)(const char *sharepath, const char *host,
-    const char *security, const char *access, void *cookie);
+	const char *security, const char *access,
+	void *cookie);
+
+/* List of hosts and their options */
+/* NOTE: share path is stored elsewhere */
+typedef struct nfs_share_list_s {
+	char host[255];
+	char opts[255];
```

Comment:
I don't see where the length of input strings are being validated. How do we protect against writing past the end of these arrays and corrupting nearby memory?